### ISSUE TITLE
[API] Cluster Templates - Member is able to reduce Access Type of other users on a cluster template

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#21720]
Version: master-head (7/23/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create a cluster template, share it with another user, make them a member.
- Share it with another member, make them an owner.
- Log in as the _member_ user, edit the cluster template.
- Notice, we can successfully change permissions on the owner user. We can limit their access to just Read-Only or remove them if we want.
- View in API - it indeed affected the clustertempalterevision.

**Result:**
Member can elevate access and change membership for any user including those with higher access.

**Comment by mrajashree** [comment]
@davidnuzik @deniseschannon @alena1108 I don't think this is a bug. If the user was given "member" access type, they have the "update" permission on the clusterTemplate resource. So they can update the Members list on the clusterTemplate resource too. Only "read-only" members won't be able to change the access type of other members.

**Comment by deniseschannon** [comment]
@mrajashree After discussing, we have agreed that this is the permission logic for members:

Ability to update templates, but explicitly exclude the ability to update membership of the template as that would provide potential privilege escalation (able to assign yourself owner permissions).

**Comment by deniseschannon** [comment]
After further discussion, we have decided that the "member" role in cluster templates doesn't make sense. We will be removing this role and keep the following 2 roles:

1. Owner - Able to CRUD templates/revisions and share CT to other people (Note: a CT owner will also need to have the global `Create Cluster Template` role in order for it to make sense, and only admins can assign this role.)
2. Read-Only - People who will be dedicated to use the cluster templates. For better UX, the UI will be updated to `User` for when assigning this role. https://github.com/rancher/rancher/issues/22200

**Comment by davidnuzik** [comment]
Version: master-head (v2.3) (8/19/19)

I can confirm now that we only have two access types to chose from in the UI when we are sharing a cluster template:

![image](https://user-images.githubusercontent.com/45179589/63308060-9e230b80-c2a5-11e9-893e-8db65a33316d.png)

I confirmed that the **Owner** member is able to create, read, update, delete the cluster template (assuming has the required `Create Cluster Template` global permission to create).

I confirmed that the **User** member is NOT able to create, update, or delete the cluster template (they can only read and leverage a cluster template to create a cluster).

I confirmed that if the User (Read-Only) member has the Create Cluster Template permission they still can't make changes to the CT that was shared with them as "User" Access Type. (They can only create their own cluster templates, as expected).
